plugins/woocommerce-blocks/assets/js/atomic/utils/blocks-registration-manager/template-change-detector.ts

```diff
@@ -3,7 +3,12 @@
  */
 import { subscribe } from '@wordpress/data';
 import { getPath, getQueryArg } from '@wordpress/url';
-import { isNumber } from '@woocommerce/types';
+
+enum ContentType {
+	POST_OR_PAGE = 'post-or-page',
+	TEMPLATE = 'template',
+	NONE = 'none',
+}
 
 interface TemplateChangeDetectorSubject {
```

Comment:
We should probably think up a new name for this since its extending beyond templates now. Perhaps `EditorChangeDetector` or `ViewChangeDetector`?

Reply:
I agree with you. Initially the idea was to have a "detector" for each content type, but I believe that having only one would be sufficient, if we think the complexity is to high, then we can separate them. I renamed it to EditorViewChangeDetector